src/Lykke.AlgoStore.CSharp.AlgoTemplate.Services/Functions/ADX/ATRFunction.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Lykke.AlgoStore.CSharp.Algo.Core.Candles;
+using Lykke.AlgoStore.CSharp.Algo.Core.Functions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Lykke.AlgoStore.CSharp.AlgoTemplate.Services.Functions.ADX
+{
+    public class ATRFunction : IFunction
+    {
+        private readonly int _period;
+        private Candle _previousInput;
+        private int _samples { get; set; }
+
+        private AtrParameters _functionParams = new AtrParameters();
+        public FunctionParamsBase FunctionParameters => _functionParams;
+
+        /// <summary>
+        /// TR - True range
+        /// </summary>
+        private double TrueRange { get; set; }
+
+        /// <summary>
+        /// True ranges for the initial calculation of Avarage True Range
+        /// </summary>
+        private Queue<double> TrueRanges { get; set; }
+
+        /// <summary>
+        /// ATR
+        /// </summary>
+        private double? AverageTrueRange { get; set; }
+
+        /// <summary>
+        /// The Previous Average True Range which will be used for the calculation of the new one
+        /// </summary>
+        private double? PreviousAverageTrueRange { get; set; }
+
+
+        public ATRFunction(AtrParameters adxParameters)
+        {
+            _period = adxParameters.Period;
+            _functionParams = adxParameters;
+            TrueRanges = _functionParams.Period == 0 ? new Queue<double>() : new Queue<double>(_functionParams.Period);
+            AverageTrueRange = 0.0d;
+        }
+
+        public double? WarmUp(IList<Candle> values)
+        {
+            if (values == null)
+                throw new ArgumentException();
+
+            foreach (var value in values)
+            {
+                _samples++;
+
+                TrueRange = ComputeTrueRange(value);
+                if (values.IndexOf(value) > 0)
+                {
+                    TrueRanges.Enqueue(TrueRange);
+                }
+
+                AverageTrueRange = CalculateAverageTrueRange();
+                _previousInput = value;
+            }
+            return AverageTrueRange;
+        }
+
+        public bool IsReady
+        {
+            get { return _samples > _period + 1; }
+        }
+
```

Comment:
Can use expression-bodied property "public bool IsReady => _samples > _period + 1;" 
You can move it on top with the other properties.